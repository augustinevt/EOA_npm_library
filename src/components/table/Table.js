import React from 'react';

class Table extends React.Component {
  constructor(props) {
    super(props);
  }

  getRows(dataObject, scopeClass) {
    return Object.keys(dataObject).map((key, i) => {
      return (
        <div className={`table__row `} key={`table-row-${i}`}>
          <div className={`table__cell key`}>{ this.props.config.keyFormatter(key) }</div>
          <div className={`table__cell  value`}>{ this.getValue(dataObject[key]) }</div>
        </div>
      );
    });
  }

  getValue(value) {
    try {
      if ( value && typeof value === 'object') {
        return value.component;
      } else {
        return this.props.config.valueFormatter(value);
      }
    } catch (e) {
      return '';
    }
  }

  render() {
    const scopeClass = this.props.scopeClass;
    const data = this.props.data;
    const rows = this.getRows(data, scopeClass);

    return (
      <div className={`table`} key={`I love cows`}>
        {rows}
      </div>
    );
  }
}

Table.defaultProps = {};

export default Table;
