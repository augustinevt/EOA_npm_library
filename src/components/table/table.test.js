import React from 'react';
import renderer from 'react-test-renderer';

import Table from './Table';

const testTableData = {
  I: 'am',
  a: 'donut'
};

const testTableConfig = {
  keyFormatter: (val) => val,
  valueFormatter: (val) => val,
};

test('default table snapshot test', () => {
  const snapshot = renderer.create(
    <Table
      data={testTableData}
      config={testTableConfig}
    />
  );

  expect(snapshot).toMatchSnapshot();
});
