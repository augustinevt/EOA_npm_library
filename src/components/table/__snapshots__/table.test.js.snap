// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default table snapshot test 1`] = `
<div
  className="table"
>
  <div
    className="table__row "
  >
    <div
      className="table__cell key"
    >
      I
    </div>
    <div
      className="table__cell  value"
    >
      am
    </div>
  </div>
  <div
    className="table__row "
  >
    <div
      className="table__cell key"
    >
      a
    </div>
    <div
      className="table__cell  value"
    >
      donut
    </div>
  </div>
</div>
`;
